docs/decisions/0014-access-policy-grouping.rst

```diff
@@ -0,0 +1,40 @@
+0014 Access Policy Grouping
+###########################
+
+Status
+******
+**Pending** Feb 2024
+
+Context
+*******
+The enterprise-access service needs to allow for the flexibility to control and distribute subsidy access policies to
+custom subsets of learners. The subdivision of access policies will improve user management and budgeting from an admin
+perspective, as well as add support for personalization to the learner experience down the line. While it is not
+the responsibility of the service to house the grouping of learners, an access policy must take into account the 
+existence of related subsets when determining if a subsidy is redeemable by an individual learner.
+
+Decision
+********
+The ``SubsidyAccessPolicy`` model will adopt a new, nullable UUID field: `enterprise_group_uuid`. This field will be
+read and factored in at time of calculating redeemability of an individual policy for a learner.
+
+How the systems will interact with one another
+++++++++++++++++++++++++++++++++++++++++++++++
+Upon provisioning a new access policy budget for a customer, the service will make a POST request to edx-platform to
+create a new ``EnterpriseGroup`` record. On successful response, the enterprise-access service will write the returned
+UUID of the newly created group to the access policy `enterprise_group_uuid` field.
+
+``SubsidyAccessPolicy``'s `can_redeem()` method already makes a request to edx-platform for
+`enterprise_contains_learner()` in which `lms_user_id`` and `enterprise_customer_uuid` are provided to confirm
+a learner's membership with the associated organization. Now, `enterprise_group_uuid` will be optionally supplied by
```

Comment:
Again, wouldn't this not be optional? We would have to know if the user's group permissions have access to this course

Reply:
optional in that I think it should be backwards compatible

Reply:
so if a policy doesn't have a group it won't blow up